controllers/roverController.ts

```diff
@@ -0,0 +1,16 @@
+import nasaClient from "../clients/nasaClient"
+import Rover from "../models/roverData"
+
+export const RoverController = {
+    Index: async (req: any, res: any) => {
+        try {
+            const roverData: Rover[] | void = await nasaClient.getRoverList()
+            const roverNames: string[] = []
+            roverData!.forEach((rover: Rover) => roverNames.push(rover.name))
```

Comment:
Generally if we find ourselves using the `!` to _tell_ typescript that something exists, its an opportunity to consider a refactor to make it so TypeScript doesn't need to be told something exists.

In this case:
- if your `getRoverList()` function will throw an error if it fails, but otherwise return the list, then its return type should not include undefined/null. This means that in your code, roverData will always be _something_, and if its not, it will get captured in the try/catch
- Alternatively your roverList function could do its own error capturing and return an empty list, then in this function you can check if you get an empty list, before accessing it

Reply:
Thank you, I've interpreted this to mean I should return the error, not just `console.error` it? With this I can now remove the `!` without any issues.